peering/models/models.py

```diff
@@ -589,6 +602,27 @@ def get_relationship_html(self):
             f'<span class="badge {badge_type}">{self.get_relationship_display()}</span>'
         )
 
+    def generate_service_ref(self) -> str:
+        """
+        Generate a Unique Service Reference for an Direct BGP Session
+        from Local ASN with 6 digit hex UUID. Example: D9268-E2E162S/D<asn>-<hex>S
+
+        Return:
+            str: Service Reference
+        """
+        asn = self.local_autonomous_system.asn
+        return "D{0}-{1}S".format(asn, uuid.uuid4().hex[:6].upper())
+
+    def save(self, *args, **kwargs):
+        """
+        Overwrite Model Save to generate a Unique Service Reference
+        if left blank on Save/Update
+        """
+        if self.service_reference is None:
```

Comment:
Since the service reference is a string, `if not self.service_reference` should be enough, checking empty string values as well.